docs/source/architecture.rst

```diff
@@ -1,17 +1,134 @@
-*****************
-PBFT Architecture
-*****************
+Architecture
+************
 
-The Sawtooth PBFT algorithm provides practical Byzantine fault tolerant (PBFT)
-consensus for Hyperledger Sawtooth. All nodes on the network participate in the
-consensus process by exchanging messages. One node, called the :term:`primary`,
-is the leader. All other nodes are called :term:`secondary nodes <secondaries>`.
+The Sawtooth PBFT algorithm is a voting-based consensus algorithm with Byzantine
+fault tolerance, which ensures `safety and liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__.
+The network can tolerate a certain number of "bad" nodes. As long as this number
+is not exceeded, the network will work properly. In addition, blocks committed
+by nodes are final, so there are no forks in the network.
+
+The nodes on the network send many messages to reach consensus, commit blocks,
+and maintain a healthy leader node, called a `primary node`. The network
+switches to a new primary (called a `view change`) at regular intervals, as well
+as when the current primary is faulty.
+
+* The primary node builds and publishes blocks.
+
+* The other nodes (called `secondary nodes`) vote on the new blocks and send
```

Comment:
I would reword this to just be "vote on blocks" instead of "vote on the new blocks". Also, I think "send messages if the primary appears unreachable or malicious" is too vague in that "send messages" doesn't really tell you anything about what they're doing if the primary is unreachable or malicious (they send messages for just about everything). Not totally sure what the best way to word this is, though.

Reply:
How about "The other nodes ... vote on new blocks and the health of the leader." ?